### PR TITLE
8255457: Shenandoah: cleanup ShenandoahMarkTask

### PR DESCRIPTION
**Pull request by shipilev**
The code in ObjArrayChunkedTask is rough around the edges, and can be cleaned up. It would be further extended in JDK-8254315, so it makes sense to make the change before the bulk of JDK-8254315 arrives.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Linux x86 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- |
| Build | ✔️ (5/5 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) | ✔️ (9/9 passed) | ❌ (1/9 failed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) |

**Failed test task**
- [Linux x86 (hs/tier1 gc)](https://github.com/shipilev/jdk/runs/1316878597)

### Issue
 * [JDK-8255457](https://bugs.openjdk.java.net/browse/JDK-8255457): Shenandoah: cleanup ShenandoahMarkTask

### Reviewers
 * [Roman Kennke](https://openjdk.java.net/census#rkennke) (@rkennke - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/883/head:pull/883`
`$ git checkout pull/883`

### CODE REVIEW

File: src/hotspot/share/gc/shenandoah/shenandoahTaskqueue.hpp [hunk 1]
```diff
@@ -169,8 +159,12 @@ class ObjArrayChunkedTask
     return (int) ((val >> pow_shift) & right_n_bits(pow_bits));
   }
 
+  inline bool decode_not_chunked(uintptr_t val) const {
+    return (val & ~right_n_bits(oop_bits + pow_bits)) == 0;
+  }
+
   inline uintptr_t encode_oop(oop obj) const {
-    return ((uintptr_t)(void*) obj) << oop_shift;
+    return ((uintptr_t) (void *) obj) << oop_shift;
```
Comment (rkennke) [inline src/hotspot/share/gc/shenandoah/shenandoahTaskqueue.hpp:167]: Looks like an extra space sneaked in here. Or was that intentional?

File: src/hotspot/share/gc/shenandoah/shenandoahTaskqueue.hpp [hunk 2]
```diff
@@ -181,12 +175,42 @@ class ObjArrayChunkedTask
...
+
+public:
+  inline oop  obj()            const { return decode_oop(_obj);   }
+  inline int  chunk()          const { return decode_chunk(_obj); }
+  inline int  pow()            const { return decode_pow(_obj);   }
+
+  inline bool is_not_chunked() const { return decode_not_chunked(_obj); }
 
-  DEBUG_ONLY(bool is_valid() const); // Tasks to be pushed/popped must be valid.
+  DEBUG_ONLY(bool is_valid() const;) // Tasks to be pushed/popped must be valid.
```
Comment (rkennke) [inline src/hotspot/share/gc/shenandoah/shenandoahTaskqueue.hpp:213]: That looks weird. I guess I prefer the previous notation, but it's your call.

**Review by rkennke — APPROVED** [review]
Looks ok. Some minor whitespace/formatting nits, but I leave it to you if/what to do about it.

File: src/hotspot/share/gc/shenandoah/shenandoahTaskqueue.hpp (hunk 2 above)
Comment (shipilev) [reply src/hotspot/share/gc/shenandoah/shenandoahTaskqueue.hpp:213]: Actually, the previous version looks weird (and CLion highlights it): `DEBUG_ONLY` is a macro that removes all contents, so `;` is left dangling in release mode.

File: src/hotspot/share/gc/shenandoah/shenandoahTaskqueue.hpp (hunk 1 above)
Comment (shipilev) [reply src/hotspot/share/gc/shenandoah/shenandoahTaskqueue.hpp]: Not intentional, fixed.

**Review by rkennke — APPROVED** [review]
Looks good to me!

**Pull request closed without merging**